src/vulkan/pipeline.cc

```diff
@@ -296,15 +304,17 @@ Result Pipeline::AddDescriptor(const BufferCommand* cmd) {
         "and binding");
   }
 
-  auto* buf_desc = static_cast<BufferDescriptor*>(desc);
-  if (cmd->GetValues().empty()) {
-    Result r = buf_desc->SetSizeInElements(cmd->GetBuffer()->ElementCount());
-    if (!r.IsSuccess())
-      return r;
-  } else {
-    Result r = buf_desc->AddToBuffer(cmd->GetValues(), cmd->GetOffset());
-    if (!r.IsSuccess())
-      return r;
+  if (desc->IsStorageBuffer() || desc->IsUniformBuffer()) {
```

Comment:
Is the assumption here that htere is no buffer data for image buffers? I don't think that's a valid assumption to make as we may have image data input, yea?